packages/fluentui/docs/src/components/Sidebar/Sidebar.tsx

```diff
@@ -31,586 +32,566 @@ const behaviorMenu: ComponentMenuItem[] = require('../../behaviorMenu');
+          content: 'Performance',
           as: NavLink,
-          to: '/prototype-table',
+          activeClassName: 'active',
+          to: '/performance',
         },
-        public: true,
-      },
-      {
-        key: 'virtualized-table',
-        title: { content: 'VirtualizedTable', as: NavLink, to: '/virtualized-table' },
-        public: true,
       },
       {
-        key: 'unstable-datepicker',
-        title: { content: 'Datepicker', as: NavLink, to: '/unstable-datepicker' },
-        public: true,
+        id: 'debugging',
+        title: { content: 'Debugging', as: NavLink, activeClassName: 'active', to: '/debugging' },
       },
-    ];
-
-    const componentTreeSection = {
-      key: 'components',
-      title: 'Components',
-      items: treeItemsByType[0].items,
-    };
+    ],
+  },
+  {
+    id: 'components',
+    title: 'Components',
+    items: treeItemsByType[0].items,
+  },
+];
+
+const changeLogUrl: string = `${config.repoURL}/blob/master/packages/fluentui/CHANGELOG.md`;
+
+const removePublicTags = prototyptesTreeItems => {
+  return prototyptesTreeItems.map(p => {
+    delete p.public;
+    return p;
+  });
+};
+
+const Sidebar: React.FC<RouteComponentProps & SidebarProps> = props => {
+  const [query, setQuery] = React.useState('');
+  const [activeItemIds, setActiveItemIds] = React.useState<string[]>([]);
+  const searchInputRef = React.useRef<HTMLInputElement>();
+  const regexQuery = React.useMemo(() => new RegExp(`.*${_.escapeRegExp(query)}`, 'i'), [query]);
+
+  const handleDocumentKeyDown = React.useCallback(
+    e => {
+      const code = getCode(e);
+      const isAZ = code >= 65 && code <= 90;
+      const hasModifier = e.altKey || e.ctrlKey || e.metaKey;
+      const bodyHasFocus = document.activeElement === document.body;
+
+      if (!hasModifier && isAZ && bodyHasFocus) searchInputRef.current?.focus();
+    },
+    [searchInputRef],
+  );
+
+  const findActiveCategoryId = (at: string, sections: ShorthandValue<any>[]): string => {
+    let newAt = at;
+    if (at.startsWith('/components')) {
+      newAt = newAt.replace(/[^\/]*$/, '');
+    }
+    if (newAt[newAt.length - 1] === '/') {
+      newAt = newAt.substr(0, newAt.length - 1);
+    }
 
-    const treeItems = this.getTreeItems();
-    const withComponents = treeItems.concat(componentTreeSection);
-    return this.getSectionsWithPrototypeSectionIfApplicable(withComponents, prototypesTreeItems);
+    return sections.find(section => {
+      return section.items.some(item => item.title?.to.startsWith(newAt));
+    })?.id;
   };
 
-  render() {
-    const sidebarStyles: ICSSInJSStyle = {
-      background: '#201f1f',
-      width: `${this.props.width}px`,
-      position: 'fixed',
-      overflowY: 'scroll',
-      top: 0,
-      left: 0,
-      padding: 0,
-      height: '100%',
-      zIndex: 1000,
-    };
+  const getSectionsWithPrototypeSectionIfApplicable = React.useCallback((currentSections, allPrototypes) => {
```

Comment:
Can we move out this function from render?

Reply:
Sure, moved.